### ISSUE TITLE
How to restore frames?

### ISSUE THREAD

**Issue by RidaAyed** [syl20bnr/spacemacs#13423]
# tldr: How can I restore two frames? One buffer on the left and one the right.

![img](https://i.imgur.com/QRhgpDq.png?1)

<a id="org7a2888d"></a>

# Steps to reproduce

1.  I open a buffer on the left window
2.  I open a shell (ansi-term) on a right window
3.  I type a type command into the  shell
4.  I reboot and restart emacs
5.  I have to repeat all steps again

# Expected outcome

A couple of key-presses restore:

1.  One buffer on the left
2.  One  shell on the right
3.  A command running in the  shell on the right

# Tried so far

## psession [fails]

The documentation lacks of a usage section, anyway let's try and error:

1.  Install psession from melpa <https://github.com/thierryvolpiatto/psession>
2.  [Steps to reproduce](#org7a2888d)
3.  `M-x` `psession-mode`
4.  `M-x` `psession-save-winconf`
5.  Input `mydesktop` `RET`
6.  `M-x` `psession-restore-winconf`
7.  Where is `mydesktop` ? > Fail

## desktop.el [fails]

Setup in `init.el`

    (require 'desktop)
    (setq desktop-save 1
          desktop-load-locked-desktop t
          desktop-dirname user-emacs-directory
          desktop-restore-frames nil
          ;; Don't save remote files and/or *gpg files.
          desktop-files-not-to-save "\\(^/[^/:]*:\\|(ftp)$\\)\\|\\(\\.gpg$\\)")
    (desktop-save-mode 1)

1.  `desktop-save`
2.  Enter: `/home/ra/.emacs.d/desktops/myDesktop/.emacs.desktop.lock`
3.  `desktop-read`
    **Desktop: 1 buffer restored, 1 failed to restore.**
4.  Or sometimes if repeating `desktop-read`
    **Desktop: 2 buffers restored.** (No frames/windows restored though)

## desktop+ [fails]

<https://github.com/ffevotte/desktop-plus>

init.el

    dotspacemacs-additional-packages '(
                                       desktop+
                                       )

1.  `desktop+-create`
2.  Enter: `/home/ra/.emacs.d/desktops/myDesktop/.emacs.desktop.lock`
3.  `desktop+-restore`
    **Desktop: 2 buffers restored.** (No frames/windows restored though)

I've already opened an issue there: <https://github.com/ffevotte/desktop-plus/issues/25> 

## purpose [fails]

<https://github.com/bmag/emacs-purpose>
Restores frames or windows though not buffers

# Possible causes

1.  Am I using the wrong shell?
2.  Is spacemacs the problem?
3.  Is exwm the problem?

# Dotfiles (excerpts)

`~.xinitrc`

    exec dbus-launch --exit-with-session emacs

`.init.el`

    (require 'exwm)
    (require 'exwm-config)
    (require 'exwm-systemtray)
    (exwm-systemtray-enable)
    (exwm-enable)

# System Info     :computer:

-   OS: gnu/linux
-   Emacs: 26.3
-   Spacemacs: 0.300.0
-   Spacemacs branch: develop (rev. 32d1b8214)
-   Graphic display: t
-   Distribution: spacemacs
-   Editing style: vim
-   Completion: helm

**Comment by jr0cket** [comment]
You ask about restoring frames, however, you only have one frame in the screenshot.  I assume you mean restoring windows in a frame.

I use spacemacs layout and have the following set in .spacemacs configuration
```
   ;; If non-nil then the last auto saved layouts are resumed automatically upon
   ;; start. (default nil)
   dotspacemacs-auto-resume-layouts t
```
This restores window layouts and all buffers that belong to files.  Buffers that rely on processes such as a terminal are not restored as the process is killed when restarting or quitting Spacemacs.

The layouts layer also includes workspaces which also save the window positions.

I assume its possible to write elisp code to create a buffer that starts a new terminal process.
